src/components/installments/ProjectInstallments.tsx

```diff
@@ -0,0 +1,426 @@
+      const emptyInputRow = {
+        type: '',
+        unit: '',
+        sum: '',
+        percentage_specifier: '',
+        account_number: '',
+        due_date: '',
+      } as InputRow;
+
+      // Create an array with a length of InstallmentTypes ENUMs.
+      // Initially fill all array items with emptyInputRow objects
+      const initialInputRows = [...new Array(Object.keys(InstallmentTypes).length)].map(() => ({ ...emptyInputRow }));
+
+      // Create a copy of initial input rows
+      const installmentRows = [...initialInputRows];
+
+      // Loop through saved installments and replace empty rows with installment data
+      installments.forEach((installment, index: number) => {
+        installmentRows[index].type = installment.type;
+        installmentRows[index].account_number = installment.account_number;
+        // Always show Finnish locale values in the date input field
+        installmentRows[index].due_date =
+          installment.due_date !== null ? moment(installment.due_date, 'YYYY-MM-DD').format('D.M.YYYY') : '';
+
+        if (installment.percentage_specifier) {
+          installmentRows[index].sum = installment.percentage === undefined ? '' : installment.percentage;
+          installmentRows[index].percentage_specifier = installment.percentage_specifier;
+          installmentRows[index].unit = unitOptions.UNIT_AS_PERCENTAGE;
+        } else {
+          installmentRows[index].sum = installment.amount === undefined ? '' : (installment.amount / 100).toFixed(2);
+          installmentRows[index].unit = unitOptions.UNIT_AS_EURO;
+        }
+      });
+
+      // Sort filled input rows in the order of the InstallmentTypes ENUM list and leave empty rows last
+      const sortedInputRows = () => {
+        const InstallmentOrder = Object.values(InstallmentTypes);
+        return installmentRows.sort((a, b) =>
+          a.type
+            ? b.type
+              ? InstallmentOrder.indexOf(a.type as InstallmentTypes) -
+                InstallmentOrder.indexOf(b.type as InstallmentTypes)
+              : -1
+            : 1
+        );
+      };
+
+      // Set initial inputRows after fetching all the saved installments
+      setInputFields(sortedInputRows);
+    }
+  }, [installments]);
+
+  // // Set data to be sent to the API
+  useEffect(() => {
+    // Create a copy of inputFields
+    const inputs = [...inputFields];
+
+    // Filter out empty rows.
+    // Row is considered as filled if any of row.type, row.account_number or row.sum has a value
+    const nonEmptyRows = inputs.filter((row) => row.type !== '' || row.account_number !== '' || row.sum !== '');
+
+    const getFormattedSum = (sum: string, asPercentage: boolean, index: number) => {
+      const replaced = sum.replaceAll(',', '.'); // convert all commas to dots
+      try {
+        const value = new Decimal(replaced);
+        if (asPercentage) {
+          return value.toString();
+        }
+        return parseFloat(value.toFixed(2)) * 100;
```

Comment:
Now that there is the proper decimal lib in use, it would be better to use it for these. I guess this would be something like `return value.mul(100).round().toNumber();`.